Fix bundle layout after self-update

robmen opened the pull request:
During layout Burn will now use the handle to source engine to copy the
bundle executable since the file may have been deleted when run as part
of a self-update.

barnson commented on src/libs/dutil/fileutil.cpp:
```diff
@@ -1058,6 +1058,134 @@ extern "C" HRESULT DAPI FileCopyUsingHandles(
+            if (lpProgressRoutine)
+            {
+                liTotalCopied.QuadPart = cbTotalCopied;
+                dwResult = lpProgressRoutine(liSourceSize, liTotalCopied, liZero, liZero, 0, CALLBACK_CHUNK_FINISHED, hSource, hTarget, lpData);
```
The duplication bugs me but yeah...

barnson commented on src/libs/dutil/fileutil.cpp:
```diff
@@ -4,8 +4,8 @@
 
 // constants
 
-const BYTE UTF8BOM[] = {0xEF, 0xBB, 0xBF};
```
.editorconfig battle commence!

robmen committed:
Fix bundle layout after self-update

During layout Burn will now use the handle to source engine to copy the
bundle executable since the file may have been deleted when run as part
of a self-update.
